stdlib/REPL/src/REPLCompletions.jl

```diff
@@ -565,13 +565,20 @@ function repl_eval_ex(@nospecialize(ex), context_module::Module)
     interp = REPLInterpreter(result)
     frame = CC.InferenceState(result, src, #=cache=#:no, interp)::CC.InferenceState
 
-    CC.typeinf(interp, frame)
+    # NOTE Use the fixed world here to make `REPLInterpreter` robust against
+    #      potential invalidations of `Core.Compiler` methods.
+    Base.invoke_in_world(COMPLETION_WORLD, CC.typeinf, interp, frame)
 
     result = frame.result.result
     result === Union{} && return nothing # for whatever reason, callers expect this as the Bottom and/or Top type instead
     return result
 end
 
+# generate code cache for `REPLInterpreter`
+code_typed(CC.typeinf, (REPLInterpreter, CC.InferenceState))
+
+const COMPLETION_WORLD = Base.get_world_counter()
```

Comment:
I thought you weren't allowed to use a world that is created during precompilation.

Reply:
@Keno mentioned that this is valid because the world age progresses linearly during sysimage generation, while it forms a tree during pkgimage precompilation. So it is fine to use the precompilation world age here while it is invalid to utilize a precompilation world in general.

Reply:
it is a problem that this means we can never remove REPL from the sysimage though

Reply:
perhaps loading during `__init__` might be an okay compromise?

Reply:
I think we need to have cache for `REPLInterpreter` available at the world age, so it will incur a startup time regression? Creating code cache for `REPLInterpreter` takes several time (and thus making it robust against invalidation is benefitial).

Reply:
Okay, I have now confirmed that the compromise works. The code cache generated during REPL precompilation is available at the time of `__init__` (assuming there are no invalidations of `Core.Compiler` methods before loading REPL, which is likely true almost always). Thank you for your suggestion.